src/twisted/web/http.py

```diff
@@ -377,6 +379,75 @@ def parseContentRange(header):
 
 
 
+class _IDeprecatedHTTPChannelToRequestInterface(Interface):
+    """
+    The interface L{HTTPChannel} expects of L{Request}.
```

Comment:
> Also, a reference in the other direction might be handy. That is, in the HTTPChannel documentation for its own requestFactory attribute - saying it must create providers of this interface.

so a subsequent PR will introduce a new streaming interface, making a liar out of that docstring.  but i'll cross that bridge when i come to it!